pkg/combustion/rpm.go

```diff
@@ -32,13 +35,42 @@ func configureRPMs(ctx *image.Context) ([]string, error) {
 
 	zap.L().Info("Configuring RPM component...")
 
-	var rpmDir string
+	packages := &ctx.ImageDefinition.OperatingSystem.Packages
+	if packages.NoGPGCheck {
+		log.Auditf("WARNING: Running EIB with disabled GPG validation is intended for development purposes only")
+		zap.S().Warn("Disabling GPG validation for the EIB RPM resolver")
+	}
+
+	var localRPMConfig *image.LocalRPMConfig
 	if isComponentConfigured(ctx, userRPMsDir) {
-		rpmDir = generateComponentPath(ctx, userRPMsDir)
+		rpmDir := generateComponentPath(ctx, userRPMsDir)
+		localRPMConfig = &image.LocalRPMConfig{
+			RPMPath: rpmDir,
+		}
+
+		gpgPath := filepath.Join(rpmDir, userGPGsDir)
+		_, err := os.Stat(gpgPath)
+		switch {
+		case err == nil:
+			if !packages.NoGPGCheck {
+				localRPMConfig.GPGKeysPath = gpgPath
+			} else {
+				log.AuditComponentFailed(rpmComponentName)
+				return nil, fmt.Errorf("found existing '%s' directory, but GPG validation is disabled", userGPGsDir)
+			}
+		case errors.Is(err, fs.ErrNotExist):
+			if !packages.NoGPGCheck {
+				log.AuditComponentFailed(rpmComponentName)
+				return nil, fmt.Errorf("GPG validation is enabled, but '%s' directory is missing", userGPGsDir)
```

Comment:
We should likely add an additional clarifying warning here that because they're side-loading RPM's, this check is triggered. It's not there if the user is simply adding repo's or trying to install packages via the list.

```suggestion
				return nil, fmt.Errorf("GPG validation is enabled, but '%s' directory is missing for side-loaded RPMs", userGPGsDir)
```